components/sts/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -80,6 +85,72 @@
      */
     private static final long serialVersionUID = 1927253892844132565L;
 
+    private String stsRedirectPage = null;
+
+    /**
+     * This method reads Passive STS Html Redirect file content.
+     * This should have been implemented in the backend but done in the front end to avoid API changes
+     *
+     * @return Passive STS Html Redirect Page File Content
+     */
+    private String readPassiveSTSHtmlRedirectPage() {
+        String redirectHtmlFilePath = CarbonUtils.getCarbonHome() + File.separator + "repository"
```

Comment:
Might be a good idea to initialize this particular variable off the critical path.